File: mobile/lib/dtos/product_variant_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'product_variant_dto.g.dart';

@JsonSerializable()
class ProductVariantDTO {
  String? id;
  String? productId;
  double? originalPrice;
  double? sellingPrice;
  String? title;
  Uri? image;

  ProductVariantDTO({
    this.id,
    this.productId,
    this.originalPrice,
    this.sellingPrice,
    this.title,
    this.image,
  });

  factory ProductVariantDTO.fromJson(Map<String, dynamic> json) =>
      _$ProductVariantDTOFromJson(json);

  Map<String, dynamic> toJson() => _$ProductVariantDTOToJson(this);
}

File: bff/src/main/kotlin/me/lucasbatista/vienna/sdk/dto/ProductVariantDTO.kt
package me.lucasbatista.vienna.sdk.dto

import java.net.URL

data class ProductVariantDTO(
    val id: String? = null,
    val productId: String? = null,
    val originalPrice: Double? = null,
    val sellingPrice: Double? = null,
    val title: String? = null,
    val image: URL? = null,
)